### PR TITLE
tweak(scripting-mono-v2): bump pilot date

### PR DESCRIPTION
**Pull request by Nova-px**
Bump pilot date to 2023-11-31 because it's currently outdated

### CODE REVIEW

File: code/components/citizen-scripting-mono-v2/src/MonoScriptRuntime.cpp [hunk 1]
```diff
@@ -263,11 +263,12 @@ int MonoScriptRuntime::HandlesFile(char* filename, IScriptHostWithResourceData*
 	}
 
 	// last supported date for this pilot of mono_rt2, in UTC
-	constexpr int maxYear = 2023, maxMonth = 8, maxDay = 31;
+	constexpr int maxYear = 2023, maxMonth = 11, maxDay = 31;
```
Comment (thorium-cfx) [inline code/components/citizen-scripting-mono-v2/src/MonoScriptRuntime.cpp:266]: November has 30 days 😏 Let's put this to December (12) instead, too much going on right now.

**Review by thorium-cfx — CHANGES_REQUESTED** [review]
Some date/day issues

**Comment by blattersturm** [comment]
Replacing the first commit in this PR might be better - this is now a bit tricky to merge as it'll contain both commits.

**Comment by AvarianKnight** [comment]
Is it a good idea to put the pilot day during the middle of the holidays?

Also, what is the point of date locking this vs just having it under an experimental flag (kind of like rdr), as while breakages are expected, it would be a better experience for the end user to  fix their resource for any breaking changes vs the entire runtime being disabled randomly. 

Possible flag:

`use_experimental_mono_v2_runtime 'I acknowledge that the mono v2 runtime will experience API & ABI breakages, and that I will be responsible to fix my resource(s) when these breakages do happen. ' `

**Comment by blattersturm** [comment]
> Also, what is the point of date locking this

To prevent the situation from RDR occurring again, where this becomes a compatibility breaking point and the temporary situation is permanent.

**Comment by AvarianKnight** [comment]
> > Also, what is the point of date locking this
> 
> To prevent the situation from RDR occurring again, where this becomes a compatibility breaking point and the temporary situation turns permanent - indeed, the RDR situation isn't changeable now without breaking everything made there since 2019.
> 
> It's not only resource _authors_ being affected, it's also people buying or downloading resources written by others. Having a hard date lock means there's at least a hard deadline for resources using the preview feature to 'break', discouraging longer-term use by people who can't maintain such.

Thanks for the explanation, this makes a lot more sense when explained like that.

**Comment by DaniGP17** [comment]
Any news about this?

**Comment by thorium-cfx** [comment]
> Any news about this?

We've had some technical difficulties that had to be solved first. This one is scheduled next.

Sorry for the inconvenience.

**Review by thorium-cfx — APPROVED** [review]

**Comment by DaniGP17** [comment]
> > Any news about this?
> 
> We've had some technical difficulties that had to be solved first. This one is scheduled next.
> 
> Sorry for the inconvenience.

Oh no problem, sorry for bothering, good luck!

**Pull request merged**